src/routes/auth.routes.js

```diff
@@ -7,15 +7,25 @@ import {
 } from '../middleware/authMiddlewares';
 
 // eslint-disable-next-line object-curly-newline
-import { signup, login, changePassword, deleteUser } from '../controllers';
+import {
+  signup,
+  login,
+  changePassword,
+  deleteUser,
+  refreshUserToken,
+} from '../controllers';
+import { verifyToken } from '../middleware';
 
 const router = express.Router();
 
 router
   .route('/signup')
   .post(signupValidation, requireAllfields, validationErrors, signup);
 router.route('/login').post(loginValidation, validationErrors, login);
-router.route('/changePassword').post(requireCredentials, changePassword);
+router
+  .route('/changePassword')
+  .post(verifyToken, requireCredentials, changePassword);
 router.route('/deleteUser').delete(deleteUser);
+router.route('/refreshToken').post(verifyToken, refreshUserToken);
```

Comment:
where do you check if there is refresh token? shouldnt u check refresh token first and then verify access token?

Reply:
it seems like the logic checking actual refresh token is missing

Reply:
bc refresh token carries the information to get a new (access) token